src/Engine/IEngine.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace CleanLiving.Engine
 {
-    public interface IEngine<TTime>
+    public interface IEngine<TTime> : IDisposable
```

Comment:
Unless for the case of abusing using statements (which is wrong on an interface?), or the interface defines a property that is a disposable, then there should be no need for IDisposable on the interface. It's up to the author of the implementation to decide whether or not it needs disposing of. That said, if the rest of the code is not clever enough to check for IDisposable on the implementation and dispose correctly, then this "hint" should be applied, imo